server/models/Availablity.js

```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+
+const availabilitySchema = new mongoose.Schema({
+    userId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: "User",
+    required: true,
+    },
+    startDate: {

```

Comment:
Good start on the availability schema, please explain how you plan to use this design for our project use cases? :)

Reply:
The userID is to identify which user contains such available periods.
startDate-endDate identify a period that belongs to the user. 
Its like "many to many" model in relational dabase
Ex:
For the Mocks for available, you can see that listed one by one, which will be store as separate records.

Reply:
@JAFAJ0 The mocks have availability for days of the week, 'Monday Tuesday...and so on'. How will we implement such a use case with this design? 😄

Reply:
My idea is to store multiple available periods and assume we want to store more than one week's data. 
With the start/end date time, for instance, from 15/08/2016 10 am-18/09/2016 3:15 pm
The front end could know which day is it in a week, or I modify the get controller later to let the front end knows with a certain date.
In my opinion, when the front end fetches a specified period, the backend gathers information, sorts them, and returns it available in that week.
Your model is clear in which day is in a week, but I think that is quite ambiguous in date and time.

Reply:
@JAFAJ0 thanks for the clarification! 👍🏽